tests/t3_fs_api.py

```diff
@@ -816,6 +816,28 @@ async def test_create_open(ctx):
 
     await fsck(ctx)
 
+async def test_open_truncate(ctx):
+    (fi, inode) = await ctx.server.create(ROOT_INODE, newname(ctx), file_mode(),
+                                          os.O_RDWR, some_ctx)
+    fh = fi.fh
+    await ctx.server.write(fh, 0, random_data(int(1.5 * ctx.max_obj_size)))
+    await ctx.server.release(fh)
+    ctx.server.inodes.flush()
+
+    data = b'this data overwrites the previous data'
+    fi = await ctx.server.open(inode.st_ino, os.O_RDWR & os.O_TRUNC, some_ctx)
+    fh = fi.fh
+    await ctx.server.write(fh, 0, data)
+    await ctx.server.release(fh)
+    ctx.server.inodes.flush()
+
+    fi = await ctx.server.open(inode.st_ino, os.O_RDONLY, some_ctx)
+    fh = fi.fh
+    assert await ctx.server.read(fh, 0, len(data)) == data
```

Comment:
Better to read more than len(data), otherwise you'll not notice if the truncation did not happen :-).